src/Service.Tests/Unittests/SqlQueryExecutorUnitTests.cs

```diff
@@ -375,6 +377,75 @@ public void ValidateStreamingLogicAsync(int readDataLoops, bool exceptionExpecte
             }
         }
 
+        /// <summary>
+        /// Validates streaming logic for QueryExecutor
+        /// In this test the streaming logic for stored procedure's is tested.
+        /// The test tries to validate the streaming across different column types (Byte, string, int etc)
+        /// Max available size is set to 4 MB, getChars and getBytes are moqed to return 1MB per read.
+        /// Exception should be thrown in test cases where we go above 4MB.
+        /// </summary>
+        [DataTestMethod, TestCategory(TestCategory.MSSQL)]
+        [DataRow(4, false,
+            DisplayName = "Max available size is set to 4MB.4 data read loop iterations, 4 columns of size 1MB -> should successfully read because max-db-response-size-mb is 4MB")]
+        [DataRow(5, true,
+            DisplayName = "Max available size is set to 4MB.5 data read loop iterations, 4 columns of size 1MB and one int read of 4 bytes -> Fails to read because max-db-response-size-mb is 4MB")]
+        public void ValidateStreamingLogicForStoredProcedures(int readDataLoops, bool exceptionExpected)
+        {
+            TestHelper.SetupDatabaseEnvironment(TestCategory.MSSQL);
+            string[] columnNames = { "StringColumn1", "StringColumn2", "ByteColumn", "ByteColumn2", "IntColumn" };
+            int[] columnSize = { 1024 * 1024, 1024 * 1024, 1024 * 1024, 1024 * 1024, 4 };
```

Comment:
maybe columnNames can be supplemented with tsql data type? e.g. varchar or nvarchar? binary/varbinary?  that will help with understanding the numbers set in columnSize[].
Also units of measure for columnSize[] could be in comment. 1024 bytes * 1024 -> 1MB per column except for int column which is 4 bytes (https://learn.microsoft.com/en-us/sql/t-sql/data-types/int-bigint-smallint-and-tinyint-transact-sql?view=sql-server-ver16#:~:text=8%20bytes-,int,4%20bytes,-smallint)?

Reply:
fixed.